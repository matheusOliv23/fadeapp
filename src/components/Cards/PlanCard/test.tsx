import { render, screen } from '@testing-library/react'

import { PlanCard, PlanCardProps } from '.'

const { getByRole, queryByText, getByText } = screen

export const data: PlanCardProps[] = [
  {
    planType: 'Semanal',
    price: 9.99,
    haveOffer: false,
    image: 'http://cdn.com/photo-name.jpg',
    buttonText: 'Assine',
  },
  {
    planType: 'Mensal',
    price: 22.99,
    haveOffer: true,
    equivalentValue: 5.75,
    offer: 54,
    image: 'http://cdn.com/photo-name.jpg',
    buttonText: 'Assine',
  },
]

describe('<PlanCard />', () => {
  it('should render by default', () => {
    const { container } = render(<PlanCard {...data[0]} />)

    const heading = getByRole('heading', { name: 'Semanal' })
    const button = getByRole('button', { name: 'Assine' })
    const offer = queryByText('54%')
    const planType = getByRole('heading', { name: 'Semanal' })
    const image = container.firstChild
    const renewText = getByText('Renovação automática')
    const price = getByText('9,99')

    expect(heading).toBeInTheDocument()
    expect(button).toBeInTheDocument()
    expect(price).toBeInTheDocument()
    expect(offer).not.toBeInTheDocument()
    expect(image).toHaveStyle(
      'background-image: url(http://cdn.com/photo-name.jpg)',
    )
    expect(planType).toBeInTheDocument()
    expect(renewText).toBeInTheDocument()
  })
  it('should render with offer', () => {
    render(<PlanCard {...data[1]} />)

    const heading = getByRole('heading', { name: 'Mensal' })
    const offerPercentage = getByText('54%')
    const planType = getByRole('heading', { name: 'Mensal' })
    const price = getByText('22,99')
    const equivalentValue = getByText('(Equivalente a R$ 5,75 por semana)')

    expect(heading).toBeInTheDocument()
    expect(planType).toBeInTheDocument()
    expect(offerPercentage).toBeInTheDocument()
    expect(price).toBeInTheDocument()
    expect(equivalentValue).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = render(<PlanCard {...data[0]} />)

    expect(container).toMatchSnapshot()
  })
})
