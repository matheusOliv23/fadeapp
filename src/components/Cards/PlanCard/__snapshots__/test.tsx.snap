// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlanCard /> should match snapshot 1`] = `
<div>
  <div
    class="
relative h-full w-[18rem] md:w-[20.75rem] sm:w-full h-[32.125rem] text-white bg-no-repeat bg-contain lg:hover:-translate-y-6 lg:duration-300 lg:transition-all lg:ease-in-out
"
    style="background-image: url(http://cdn.com/photo-name.jpg);"
  >
    <div
      class="
flex h-full w-[18rem] max-w-[19rem] md:max-w-[20.75rem] flex-col items-center justify-center gap-10 text-center sm:w-[20.875rem] bg-gradient-2
"
    >
      <div
        class="
flex flex-col items-center
"
      >
        <h3
          class="
gap-2 text-6xl font-medium mb-10
"
        >
          Semanal
        </h3>
        <div
          class="
relative flex items-end justify-start gap-1 mt-5
"
        >
          <span
            class="
absolute -left-6 -top-2 justify-end font-medium text-base
"
          >
            R$
          </span>
          <p
            class="
text-7xl font-medium
"
          >
            9,99
            <span
              class="
justify-start p-1 text-lg lowercase
"
            >
              /
              Semanal
            </span>
          </p>
        </div>
      </div>
      <div
        class="
h-[2.875rem] w-[16rem] mt-3
"
      >
        <button
          class="flex items-center justify-center rounded-10 bg-primary font-bold text-secondary transition hover:text-primary hover:bg-transparent border-2 border-primary transition-color w-[16.188rem] text-2xl h-[2.875rem] "
        >
          Assine
        </button>
        <p
          class="
py-4 text-base text-sm
"
        >
          Renovação automática
        </p>
      </div>
    </div>
  </div>
</div>
`;
